fe1-web/src/features/wallet/objects/transaction/Transaction.ts

```diff
@@ -0,0 +1,381 @@
+import { Base64UrlData, Hash, KeyPair, PopToken, PublicKey } from 'core/objects';
+import { SCRIPT_TYPE, COINBASE_HASH } from 'resources/const';
+
+import { TransactionInput, TransactionInputJSON, TransactionInputState } from './TransactionInput';
+import {
+  TransactionOutput,
+  TransactionOutputJSON,
+  TransactionOutputState,
+} from './TransactionOutput';
+
+export interface TransactionJSON {
+  version: number;
+  inputs: TransactionInputJSON[];
+  outputs: TransactionOutputJSON[];
+  lock_time: number;
+}
+
+export interface TransactionState {
+  version: number;
+  inputs: TransactionInputState[];
+  outputs: TransactionOutputState[];
+  lockTime: number;
+  transactionId?: string;
+}
+
+/**
+ * A coin transaction object
+ */
+export class Transaction {
```

Comment:
I feel like this file contains a lot of lines (381). Maybe it could be split into 2 files, or reduced by using private functions for somewhat similar code (like `create` and `createCoinbase`).

Reply:
I tried to make it as small as possible but also to keep it modulable, maybe if we really want to reduce it we could split the verification process in another file, but I do not know if it is worth it